trace_replay/trace_replay.cc

```diff
@@ -341,7 +341,8 @@ Status Replayer::MultiThreadReplay(uint32_t threads_num) {
       thread_pool.Schedule(&Replayer::BGWorkIterSeekForPrev, ra, nullptr,
                            nullptr);
       ops++;
-    } else if (ra->trace_entry.type == kTraceEnd) {
+    } else {
+      // It is the case that ra->trace_entry.type == kTraceEnd
```

Comment:
Can you put a `assert(ra->trace_entry.type == kTraceEnd)` here?

Reply:
Sure.

Reply:
Or maybe keep original `else if ()`, but add an extra `else` in which assert(false).
Another thing I suggest is using unique_ptr to guard `ra`. You can avoid the `delete ra` if break out of the loop. You can pass the pointer to `Schedule` by calling the `unique_ptr::release()` funciton.

Reply:
You mean using unique_ptr and release the ownership when pass to Schedule, such that I can remove all the delete?

Reply:
Yes.